internal/consumer/qldb_fill.go

```diff
@@ -0,0 +1,237 @@
+	}
+	receiver, err := qt.Receiver.ConvertToPostgresAccount()
+	if err != nil {
+		return fmt.Errorf("unable to convert QldbAccount to Account: %w", err)
+	}
+	transaction, err := qt.ConvertToPostgresTransaction()
+	if err != nil {
+		return fmt.Errorf(
+			"unable to convert QldbTransaction to Postgres Transaction: %w",
+			err,
+		)
+	}
+
+	if err := relationaldb.InsertTransaction(
+		ctx, transaction, sender, receiver); err != nil {
+		return fmt.Errorf("unable to upsert transaction: %w", err)
+	}
+
+	fillUUID, err := uuid.Parse(f.FillId)
+	if err != nil {
+		return fmt.Errorf("unable to parse fill id into valid uuid: %w", err)
+	}
+
+	var senderAmount *ion.Decimal
+	var receiverAmount *ion.Decimal
+	var feeCommissionSenderAccountId uuid.UUID
+	if f.Side == "BUY" {
+		senderAmount = f.FilledValue
+		receiverAmount = f.FilledQuantity
+		feeCommissionSenderAccountId = sender.Id
+	} else {
+		senderAmount = f.FilledQuantity
+		receiverAmount = f.FilledValue
+		feeCommissionSenderAccountId = receiver.Id
+	}
+
+	senderEntryId := f.Metadata[relationaldb.SenderEntryKey]
+	if err := prepareAndInsertEntry(
+		ctx,
+		senderEntryId,
+		sender.Id,
+		transaction.Id,
+		senderAmount,
+		"DEBIT",
+		f.CreatedAt,
+		fillUUID,
+	); err != nil {
+		return err
+	}
+
+	receiverEntryId := f.Metadata[relationaldb.ReceiverEntryKey]
+	if err := prepareAndInsertEntry(
+		ctx,
+		receiverEntryId,
+		receiver.Id,
+		transaction.Id,
+		receiverAmount,
+		"CREDIT",
+		f.CreatedAt,
+		fillUUID,
+	); err != nil {
+		// ToDo: Delete Sender Entry
+		return err
+	}
+
+	// If Commission was taken out, write the corresponding entries
+	if commissionReceiverId, ok := f.Metadata[relationaldb.CommissionReceiverEntryKey]; ok {
+		commissionSenderId := f.Metadata[relationaldb.CommissionSenderEntryKey]
+		if err := prepareAndInsertEntry(
+			ctx,
+			commissionSenderId,
+			feeCommissionSenderAccountId,
+			transaction.Id,
+			f.Commission,
+			"DEBIT",
+			f.CreatedAt,
+			fillUUID,
+		); err != nil {
+			return err
+		}
+
+		if err := prepareAndInsertEntry(
+			ctx,
+			commissionReceiverId,
+			feeCommissionSenderAccountId,
+			transaction.Id,
+			f.Commission,
+			"CREDIT",
+			f.CreatedAt,
+			fillUUID,
+		); err != nil {
+			// ToDo: Delete Sender Entry
+			return err
+		}
+	}
+
+	// if Fee was taken out, write the corresponding entries
+	if feeReceiverId, ok := f.Metadata[relationaldb.FeeReceiverEntryKey]; ok {
+		feeSenderId := f.Metadata[relationaldb.FeeSenderEntryKey]
+		if err := prepareAndInsertEntry(
+			ctx,
+			feeSenderId,
+			feeCommissionSenderAccountId,
+			transaction.Id,
+			f.FeeAmount,
+			"DEBIT",
+			f.CreatedAt,
+			fillUUID,
+		); err != nil {
+			return err
+		}
+
+		if err := prepareAndInsertEntry(
+			ctx,
+			feeReceiverId,
+			feeCommissionSenderAccountId,
+			transaction.Id,
+			f.FeeAmount,
+			"CREDIT",
+			f.CreatedAt,
+			fillUUID,
+		); err != nil {
+			// ToDo: Delete Sender Entry
```

Comment:
Is this an open issue?

Reply:
Not yet, I'm just going to add it as DELETE is easy.